### DISCUSSION TITLE
why each trace compress separately?

### DISCUSSION THREAD

**Discussion by zdyj3170101136** [grafana/tempo#3079] [General]
as i know, the block consist of trace id and trace data.
seems that the trace data is compress separately?
this will lower the compression ratio!
why not compress the whole block data?

**Comment by joe-elliott** [comment]
this is closed but i'm going to add an answer if anyone is interested. there are 2 primary backend formats in Tempo "v2" and the new "vParquet" line (currently on vParquet3).

The v2 format supports trace by id lookup only. Using it makes Tempo a giant, hyper efficient trace by id key value store on object storage. v2 compresses traces at the a configurable page size. Generally many traces map to one page.

The vParquet formats map traces into a columnar format. Each column is compressed differently depending on the nature of the column.
